hazelcast/src/test/java/com/hazelcast/internal/memory/impl/HeapMemoryManager.java

```diff
@@ -91,12 +96,36 @@ public long allocate(long size) {
 
         @Override
         public long reallocate(long address, long currentSize, long newSize) {
-            throw new UnsupportedOperationException();
+            assertFitsInt(address);
+            assertFitsInt(currentSize);
+            assertFitsInt(newSize);
+            validateBlock(address, currentSize);
+            final long sizeDelta = newSize - currentSize;
+            final long newAddress;
+            if (address == lastAllocatedAddress) {
+                heapTop += sizeDelta;
+                newAddress = address;
+            } else {
+                newAddress = allocate(newSize);
+                System.arraycopy(storage, (int) address, storage, (int) newAddress,
+                        (int) Math.min(currentSize, newSize));
+                allocatedAddrs.remove(address);
+            }
+            usedMemory += sizeDelta;
```

Comment:
In case of reallocated address is not last allocated address, this calculation `usedMemory += sizeDelta;` seems wrong. `allocate` method itself increases the used memory size as new size so the old size must be decreased here not size delta.